pkgs/development/libraries/science/math/magma/default.nix

```diff
@@ -1,25 +1,50 @@
-{ stdenv, fetchurl, cmake, gfortran, cudatoolkit, libpthreadstubs, liblapack }:
+{ stdenv, fetchurl, cmake, gfortran, cudatoolkit, libpthreadstubs, liblapack
+, mklSupport ? false, mkl ? null
+}:
```

Comment:
You could also pick up mkl-dnn (USE_MKLDNN=ON) and protobuf (BUILD_CUSTOM_PROTOBUF=0) directly from Nix, which should reduce build time and improve interoperability between it and other Nix built libraries.

Reply:
For magma? Magma is now merged and these diffs will be removed from the PR.
---
For pytorch:

Protobuf requires more digging to see how to correctly hook it into pytorch's build process since environment variables are masked. Really this means that I need to figure out how to build the nix derivation with https://github.com/pytorch/pytorch#adjust-build-options-optional. I'm on board, but I think this should be another PR.

MKLDNN is not yet in nix, the open PR is here: https://github.com/NixOS/nixpkgs/pull/68014. It's not clear that `USE_MKLDNN` does the thing you want, however, since I believe MKLDNN is built by default if mkl is present (hence the [mkldnn hack](https://github.com/NixOS/nixpkgs/blob/187025529c4edeb42ed7e688675083097ac05ff5/pkgs/development/python-modules/pytorch/default.nix#L142)). `USE_*` flags generally just toggle, from what I've seen, and I can't find anything in https://github.com/pytorch/pytorch/blob/v1.2.0/setup.py which would allow us to set the path. Using CMake directly might, but then I'd have to figure out how to partially run `buildPythonPackage` and then run `ccmake`. w.r.t. mkldnn I'm also a little concerned in version-matching if there isn't a clean configuration in the setup.py.

Reply:
Followup: I tried both of these things and couldn't get either package to be hooked up correctly on a first pass.

Reply:
@stites Thanks.  Once MKL-DNN is available, I'll try to get a follow-up PR to you for these.  Appreciate you taking a look.

Reply:
Please do! I would actually encourage you to try it out using https://github.com/stites/pytorch-world, which is where I am drafting all the work that goes in here.